packages/ua-devtools-solana/src/oft/config.ts

```diff
@@ -43,72 +43,37 @@ const onlyEdgesFromSolana = (
 }
 
-export const initNonce: OAppConfigurator = createConfigureEdges(
-    onlyEdgesFromSolana(async ({ vector: { to } }, sdk: OFT) => {
-        const logger = createOFTLogger()
-
-        if (typeof sdk.initializeNonce !== 'function') {
-            return logger.warn(`Could not find initializeNonce() method on OFT SDK, skipping`), undefined
-        }
-
-        if (typeof sdk.isNonceInitialized !== 'function') {
-            return logger.warn(`Could not find isNonceInitialized() method on OFT SDK, skipping`), undefined
-        }
-
-        const isInitialized = await sdk.isNonceInitialized(to.eid, to.address)
-        return isInitialized ? undefined : sdk.initializeNonce(to.eid, to.address)
-    })
-)
-
-export const initSendLibrary: OAppConfigurator = createConfigureEdges(
+export const initConfig: OAppConfigurator = createConfigureEdges(
     onlyEdgesFromSolana(async ({ vector: { to } }, sdk) => {
-        const logger = createOFTLogger()
-
-        if (typeof sdk.initializeSendLibrary !== 'function') {
-            return logger.warn(`Could not find initializeSendLibrary() method on OFT SDK, skipping`), undefined
-        }
-
-        if (typeof sdk.isSendLibraryInitialized !== 'function') {
-            return logger.warn(`Could not find isSendLibraryInitialized() method on OFT SDK, skipping`), undefined
+        if (typeof sdk.initConfig !== 'function') {
+            return createOFTLogger().warn(`Could not find initConfig() method on OFT SDK, skipping`), undefined
         }
-
-        const isInitialized = await sdk.isSendLibraryInitialized(to.eid)
-        return isInitialized ? undefined : sdk.initializeSendLibrary(to.eid)
+        return sdk.initConfig(to.eid)
     })
 )
 
-export const initReceiveLibrary: OAppConfigurator = createConfigureEdges(
-    onlyEdgesFromSolana(async ({ vector: { to } }, sdk) => {
+export const addRemote: OAppConfigurator = createConfigureEdges(
+    onlyEdgesFromSolana(async ({ vector: { to } }, sdk: OFT) => {
         const logger = createOFTLogger()
 
-        if (typeof sdk.initializeReceiveLibrary !== 'function') {
-            return logger.warn(`Could not find initializeReceiveLibrary() method on OFT SDK, skipping`), undefined
+        if (typeof sdk.addRemote !== 'function') {
+            return logger.warn(`Could not find addRemote() method on OFT SDK, skipping`), undefined
         }
 
-        if (typeof sdk.isReceiveLibraryInitialized !== 'function') {
-            return logger.warn(`Could not find isReceiveLibraryInitialized() method on OFT SDK, skipping`), undefined
-        }
-
-        const isInitialized = await sdk.isReceiveLibraryInitialized(to.eid)
-        return isInitialized ? undefined : sdk.initializeReceiveLibrary(to.eid)
+        return sdk.addRemote(to.eid)
     })
 )
 
-export const initOFTConfig: OAppConfigurator = createConfigureEdges(
-    onlyEdgesFromSolana(async ({ vector: { to }, config }, sdk) => {
+export const setPeer: OAppConfigurator = createConfigureEdges(
```

Comment:
This looks like it belongs to the SDK under `setPeer` right? What we could do is have `setPeer` return `OmniTransaction | OmniTransaction[]` instead of `OmniTransaction` on the SDK and in case of Solana return two transactions if necessary. We already have this pattern for the ULN config

Reply:
Tested and done.